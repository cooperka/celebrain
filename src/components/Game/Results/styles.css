.Results img {
  height: 10em;
}

.Results .grid-container {
  display: grid;
  grid-template-columns: auto auto;
  margin: 0 auto 1rem auto;
  align-items: center;
  max-width: 50rem;
}

.Results .grid-item {
  margin-bottom: 1rem;
}

.Results .guess-text {
  margin-left: 1rem;
  font-size: 1.6rem;
  font-weight: normal;
  text-align: left;
}

.Results .guess-text.correct {
  /* Material Green 700. */
  color: #388E3C;
}

.Results .guess-text.incorrect {
  /* Material Red 700. */
  color: #D32F2F;
}
