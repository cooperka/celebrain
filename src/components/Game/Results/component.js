// @flow

import React, { Component } from 'react';
import { connect } from 'react-redux';

import { setGameState } from '../actions';
import { gameState } from '../reducers';

import './styles.css';

type Props = {
  handleStartGame: () => void,
};

class Results extends Component<Props> {

  static mapDispatchToProps(dispatch) {
    return {
      handleStartGame: () => {
        dispatch(setGameState(gameState.MEMORIZE));
      },
    };
  }

  render() {
    const { handleStartGame } = this.props;

    return (
      <div className="Results">
        <div className="title">Results</div>
        <button onClick={handleStartGame}>Try again</button>
      </div>
    );
  }

}

export default connect(undefined, Results.mapDispatchToProps)(Results);
