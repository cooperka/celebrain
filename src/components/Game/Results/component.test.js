// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import Results from './component';

it('renders', () => {
  const tree = renderer.create(<Results />);
  expect(tree.toJSON()).toMatchSnapshot();
});
