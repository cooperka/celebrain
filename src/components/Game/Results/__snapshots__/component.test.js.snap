// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="Results"
>
  <div
    className="title"
  >
    Results
  </div>
  <button
    onClick={undefined}
  >
    Try again
  </button>
</div>
`;
